src/explorer/WebAppTreeItem.ts

```diff
@@ -126,3 +127,20 @@ export class WebAppTreeItem extends SiteTreeItem {
         return await fs.readFile(templatePath, 'utf8');
     }
 }
+
+export class InvalidWebAppTreeItem implements IAzureTreeItem {
+    public static contextValue: string = 'file';
+    public readonly contextValue: string = InvalidWebAppTreeItem.contextValue;
+    public readonly commandId: string = 'appService.showFile';
+
+    constructor(readonly label: string) {
```

Comment:
If you didn't know, the contextValue is used to determine which commands are shown in the context menu. See the 'viewItem' property in package.json:
https://github.com/Microsoft/vscode-azureappservice/blob/master/package.json#L294

If you want to support OpenInPortal, just do the same thing in package.json except use this treeItem's contextValue as the viewItem instead of 'appService'. You also have to specify the `id` property on this treeItem, which is used under the covers [here](https://github.com/Microsoft/vscode-azuretools/blob/master/ui/src/treeDataProvider/AzureNode.ts#L125) for openInPortal. I don't think you'll have to completely construct the fake Site object - it should just be Site.id (which should probably exist even if serverFarmId does not - but idk)

Reply:
Yeah, site.id doesn't exist for these either, so that's not going to work. I'd love for it to just open to the "WebApps" blade in the portal, but I'm not sure how to get a direct link for that (when I go there in the portal, my tenant id is in the URL).

Reply:
I wouldn't worry about it then. Judging by these screenshots, the Portal wasn't that helpful anyways: https://github.com/Microsoft/vscode-azureappservice/pull/370#discussion_r177598043

Reply:
Roger that!